assets/app/view/game/tile_manifest.rb

```diff
@@ -12,8 +12,10 @@ def render_tile_selector(remaining, tile)
         return [] unless @tile_selector
         return [] if @tile_selector.role != :tile_page || @tile_selector.hex&.tile&.name != tile.name
 
-        upgrade_tiles = @game.all_potential_upgrades(@tile_selector.hex.tile).map do |t|
-          [t, remaining[t.name]&.any? ? nil : 'None Left']
+        upgrade_tiles = @game.all_potential_upgrades(@tile_selector.hex.tile).map do |t_info|
```

Comment:
i don't think we should have this

Reply:
I will make a simplified solution, which would remove the changes in the general code.